docs/tutorials/optimizers_cyclicallearningrate.ipynb

```diff
@@ -0,0 +1,450 @@
+        "            layers.MaxPooling2D(pool_size=(2, 2)),\n",
+        "            layers.Conv2D(32, (5, 5), activation=\"relu\"),\n",
+        "            layers.MaxPooling2D(pool_size=(2, 2)),\n",
+        "            layers.SpatialDropout2D(0.2),\n",
+        "            layers.GlobalAvgPool2D(),\n",
+        "            layers.Dense(128, activation=\"relu\"),\n",
+        "            layers.Dense(10, activation=\"softmax\"),\n",
+        "        ]\n",
+        "    )\n",
+        "    return model\n",
+        "\n",
+        "def train_model(model, optimizer):\n",
+        "    model.compile(loss=\"sparse_categorical_crossentropy\", optimizer=optimizer,\n",
+        "                       metrics=[\"accuracy\"])\n",
+        "    history = model.fit(x_train,\n",
+        "        y_train,\n",
+        "        batch_size=BATCH_SIZE,\n",
+        "        validation_data=(x_test, y_test),\n",
+        "        epochs=EPOCHS)\n",
+        "    return history"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "RlRKWRWrSk_t"
+      },
+      "source": [
+        "In the interest of reproducibility, the initial model weights are serialized which you will be using to conduct our experiments. "
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "-JxnpsIzwCgj"
+      },
+      "outputs": [],
+      "source": [
+        "initial_model = get_training_model()\n",
+        "initial_model.save(\"initial_model\")"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "oNF33-tBSuFG"
+      },
+      "source": [
+        "## Train a model without CLR"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "Q4dEJtQzwjei"
+      },
+      "outputs": [],
+      "source": [
+        "standard_model = tf.keras.models.load_model(\"initial_model\")\n",
+        "no_clr_history = train_model(standard_model, optimizer=\"sgd\")"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "eaK0PAN-Sy6l"
+      },
+      "source": [
+        "## Define CLR schedule\n",
+        "\n",
+        "The `tfa.optimizers.CyclicalLearningRate` module return a direct schedule that can be passed to an optimizer. The schedule takes a step as its input and outputs a value calculated using CLR formula as laid out in the paper. "
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "ne0b8aGNyc3v"
+      },
+      "outputs": [],
+      "source": [
+        "steps_per_epoch = len(x_train) // BATCH_SIZE\n",
+        "clr = tfa.optimizers.CyclicalLearningRate(initial_learning_rate=INIT_LR,\n",
+        "    maximal_learning_rate=MAX_LR,\n",
+        "    scale_fn=lambda x: 1/(2.**(x-1)),\n",
+        "    step_size=8 * steps_per_epoch\n",
```

Comment:
> 8 * steps_per_epoch

Maybe lower this 8 -> 2 or something  so that the example training task has a chance to run a couple of cycles (or just train a little longer).